Support per-directory history, support autopair, fix trap int stealing int signal
diff --git a/history-search-multi-word b/history-search-multi-word
index 7dcd99c..4e26d3b 100644
--- a/history-search-multi-word
+++ b/history-search-multi-word
@@ -172,8 +172,11 @@ if (( ${+functions[_hsmw_main]} == 0 )); then
     #
     # Colorify
     #
-
-    local preamble=$'\n'"Ctrl-K – context, Ctrl-J – bump. Entry #$__hsmw_hcw_index of $max_index"$'\n'
+    local history_type=""
+    if (( ${+_per_directory_history_is_global} )); then
+      history_type=", Ctrl-G – ${${_per_directory_history_is_global/false/global}/true/local}"
+    fi
+    local preamble=$'\n'"Ctrl-K – context, Ctrl-J – bump$history_type. Entry #$__hsmw_hcw_index of $max_index"$'\n'
     integer offset=${#preamble}+${#BUFFER}
     POSTDISPLAY="$preamble$text"
 
@@ -279,7 +282,7 @@ if [[ "$__hsmw_hcw_call_count" -eq "1" ]]; then
 
   # Make the hsmw keymap a copy of the current main
   bindkey -N hsmw emacs
-
+  bindkey -M hsmw "^?" backward-delete-char
   local down_widget="${${${WIDGET%-backwards}%-pbackwards}%-pforwards}"
   local up_widget="${down_widget}-backwards"
   local pdown_widget="${down_widget}-pforwards"
@@ -412,7 +415,7 @@ if [[ "$__hsmw_hcw_call_count" -eq "1" ]]; then
   trap '(( __hsmw_recedit_trap_executed == 0 )) && zle && { __hsmw_recedit_trap_executed=1; zle .send-break; }' INT
 
   if zle .recursive-edit -K hsmw; then
-    trap '' INT
+    trap - INT
 
     if [[ "$__hsmw_hcw_index" -gt "0" ]]; then
       if (( __hsmw_ctx )); then
@@ -431,7 +434,7 @@ if [[ "$__hsmw_hcw_call_count" -eq "1" ]]; then
       unset __hsmw_conc
     fi
   else
-    trap '' INT
+    trap - INT
     (( __hsmw_conc )) && BUFFER=""
     unset __hsmw_conc
   fi
